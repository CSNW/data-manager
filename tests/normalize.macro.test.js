const pluginTester = require('babel-plugin-tester');
const plugin = require('babel-plugin-macros');

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: { filename: __filename },
  tests: [
    `
      import { flatMap } from 'data-manager';
      import normalize from '../normalize.macro';

      const b = 'b';
      const d = 'd';

      flatMap(normalize({
        a: 'a',
        b,
        y: {
          columns: ['c', d],
          category: 'type'
        }
      }));
    `,
    `
      import { flatMap } from 'data-manager';
      import normalize from '../normalize.macro';

      const b = 'b';

      flatMap(normalize({
        a: 'a',
        b,
        y: {
          columns: ['c', 'd'],
          categories: {
            c: { isC: true, isD: false },
            d: { isC: false, isD: true },
          }
        }
      }));
    `,
    `
      import { flatMap } from 'data-manager';
      import normalize from '../normalize.macro';

      const b = 'b';
      const f = 'f';

      flatMap(normalize(
        {
          a: 'a',
          b,
          y: {
            columns: ['c', 'd'],
            categories: {
              c: { isC: true, isD: false },
              d: { isC: false, isD: true },
            }
          }
        },
        ['e', f]
      ));
    `,
    `
      import { flatMap } from 'data-manager';
      import normalize from '../normalize.macro';

      const b = 'b';
      const f = 'f';

      flatMap(normalize(
        {
          a: 'a',
          b,
          y: {
            columns: ['c', 'd'],
            categories: {
              c: { isC: true, isD: false },
              d: { isC: false, isD: true },
            }
          }
        },
        'e', f
      ));
    `
  ]
});
