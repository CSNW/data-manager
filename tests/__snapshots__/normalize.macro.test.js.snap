// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros 1. macros: 1. macros 1`] = `
"
import { flatMap } from 'data-manager';
import normalize from '../normalize.macro';

const b = 'b';
const d = 'd';

flatMap(normalize({
  a: 'a',
  b,
  y: {
    columns: ['c', d],
    category: 'type'
  }
}));

      ↓ ↓ ↓ ↓ ↓ ↓

import { flatMap } from 'data-manager';
const b = 'b';
const d = 'd';
flatMap(row => {
  return [{
    a: row['a'],
    b: row[b],
    y: row['c'],
    'type': 'c'
  }, {
    a: row['a'],
    b: row[b],
    y: row[d],
    'type': d
  }];
});
"
`;

exports[`macros 2. macros: 2. macros 1`] = `
"
import { flatMap } from 'data-manager';
import normalize from '../normalize.macro';

const b = 'b';

flatMap(normalize({
  a: 'a',
  b,
  y: {
    columns: ['c', 'd'],
    categories: {
      c: { isC: true, isD: false },
      d: { isC: false, isD: true },
    }
  }
}));

      ↓ ↓ ↓ ↓ ↓ ↓

import { flatMap } from 'data-manager';
const b = 'b';
flatMap(row => {
  return [{
    a: row['a'],
    b: row[b],
    y: row['c'],
    isC: true,
    isD: false
  }, {
    a: row['a'],
    b: row[b],
    y: row['d'],
    isC: false,
    isD: true
  }];
});
"
`;
